gtsfm/loader/yfcc_imb_loader.py

```diff
@@ -0,0 +1,193 @@
+"""Loader for the Image-Matching-Benchmark's YFCC dataset.
+
+References: https://www.cs.ubc.ca/research/image-matching-challenge/
+
+Authors: Ayush Baid
+"""
+import os.path as osp
+from typing import List, Optional
+
+import numpy as np
+from gtsam import Cal3Bundler, PinholeCameraCal3Bundler, Pose3, Rot3
+
+import gtsfm.utils.io as io_utils
+import thirdparty.colmap.read_write_model as colmap_read_write_model
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+class YfccImbLoader(LoaderBase):
+    """Loader for IMB's YFCC dataset."""
+
+    def __init__(self, folder: str, coviz_thresh: float = 0.1):
+        """Initializes the loader.
+
+        Args:
+            folder: the base folder of the dataset.
+            coviz_thresh (optional): threshold for covisibility between two
+                                     images to be considered a valid pair.
+                                     Defaults to 0.1.
+        """
+
+        self._folder = folder
+
+        # load all the image pairs according to the covisibility threshold used
+        # in IMB's reporting
+        visibility_file = osp.join(
+            self._folder,
+            "new-vis-pairs",
```

Comment:
does the "new-vis-pairs" terminology come from IMB @ayushbaid ?

Reply:
yes. Its their variable naming convention. From what I understand, they did their own co-visibility check and are not relying on the original YFCC dataset's information